src/_guides/language/effective-dart/design_migrated.md

```diff
@@ -933,20 +939,33 @@ not intended to be invoked from Dart code and don't need a corresponding getter.
 [angular]: {{site.angulardart}}
 
 
-### AVOID returning `null` from members whose return type is `bool`, `double`, `int`, or `num`.
+### AVOID public `late final` fields without initializers.
+
+Unlike other `final` fields, a `late final` field without an initializer *does*
+define a setter. If that field is public, then the setter is public. This is
+rarely what you want. Fields are usually marked late so that they can be
+initialized *internally* at some point in the instance's lifetime, often inside
+the constructor body.
+
+Unless you do want users to call the setter, it's better to either:
+
+* not use `late`,
+* ensure the `late` field is initialized at its declaration,
+* or make the `late` field private and define a public getter to access it.
```

Comment:
So much easier to read! Some further tweaks... Apparently, "either <3 things>" is controversial (https://developers.google.com/style/word-list#letter-e). Also, list items should be parallel. So maybe:

```suggestion
Unless you *do* want users to call the setter, it's better to pick one of the
following solutions:

* Don't use `late`.
* Use `late`, but initialize the `late` field at its declaration.
* Use `late`, but make the `late` field private and define a public getter for it.
```